kmath-nd4j/src/main/kotlin/space/kscience/kmath/nd4j/arrays.kt

```diff
@@ -5,4 +5,6 @@
 
 package space.kscience.kmath.nd4j
 
-internal fun LongArray.toIntArray(): IntArray = IntArray(size) { this[it].toInt() }
+import space.kscience.kmath.misc.toIntExact
+
+internal fun LongArray.toIntArray(): IntArray = IntArray(size) { this[it].toIntExact() }
```

Comment:
Is it used somewhere? Does not seem to be safe.

Reply:
It is used for converting ND4J `long[]` indices to KMath's `int[]`.